openquake/hazardlib/sourcewriter.py

```diff
@@ -444,6 +444,8 @@ def get_source_attributes(source):
     """
     attrs = {"id": source.source_id, "name": source.name}
     if isinstance(source, NonParametricSeismicSource):
+        if not hasattr(source.data[0][0], 'weight'):
+            return attrs
```

Comment:
Can you add a comment explaining when this happens?

Reply:
Hard to explain. It's occurring when creating the sources for the JPN model.

Reply:
The weight is actually something you use ;)